homework/index.js

```diff
@@ -21,27 +21,117 @@
+      createAndAppend('td', description, {
+        html: 'Description:',
+        class: 'label',
+      });
+      createAndAppend('td', description, { html: repository.description });
+    }
+    const forks = createAndAppend('tr', tableBody);
+    createAndAppend('td', forks, { html: 'Forks:', class: 'label' });
+    createAndAppend('td', forks, { html: repository.forks });
+    const update = createAndAppend('tr', tableBody);
+    createAndAppend('td', update, { html: 'Updated:', class: 'label' });
+    createAndAppend('td', update, { html: repository.updated_at });
+  }
+
+  function showContributors(rightDiv, repository) {
+    rightDiv.innerHTML = '';
+    createAndAppend('p', rightDiv, {
+      html: 'Contributions',
+      class: 'contributor-head',
+    });
+    fetchJSON(repository.contributors_url, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', rightDiv, {
+          html: err.message,
+          class: 'alert-error',
+        });
+      }
+
+      contributors.map(contributor => {
+        const contributorUL = createAndAppend('ul', rightDiv, {
+          class: 'contributor-ul',
+        });
+        const contributorLi = createAndAppend('li', contributorUL, {
+          class: 'contributor-li',
+        });
+
+        const contributorLink = createAndAppend('a', contributorLi, {
+          target: '_blank',
+          href: contributor.html_url,
+        });
+        createAndAppend('img', contributorLink, {
+          class: 'contributor-image',
+          src: contributor.avatar_url,
+        });
+        const contributorDetails = createAndAppend('div', contributorLink, {
+          class: 'contributor-details',
+        });
+
+        createAndAppend('div', contributorDetails, {
+          html: contributor.login,
+          class: 'contributor-names',
+        });
+        createAndAppend('div', contributorDetails, {
+          html: contributor.contributions,
+          class: 'contributors',
+        });
+      });
+    });
+  }
   function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { html: 'HYF Repositories', class: 'hyf' });
+    const select = createAndAppend('select', header, {
+      class: 'select',
+      id: 'list',
+    });
+    const container = createAndAppend('div', root, { class: 'container' });
+    const left = createAndAppend('div', container, { class: 'left_div' });
+    const right = createAndAppend('div', container, { class: 'right_div' });
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        createAndAppend('div', root, {
+          html: err.message,
+          class: 'alert-error',
+        });
```

Comment:
If there is an error you should not try and execute the rest of the code because there is no data to work on. You should add a `return` statement at the end of this `if` block to return 'early' from this function. 

This is what I see in the console when I and run your code and introduce an error in the url:

```
index.js:124 Uncaught TypeError: Cannot read property 'sort' of undefined
    at fetchJSON (index.js:124)
    at XMLHttpRequest.xhr.onload (index.js:12)
```